boefjes/plugins/kat_censys/normalize.py

```diff
@@ -1,16 +1,45 @@
+import ipaddress
 import json
 from typing import Iterator, Union
 
 from octopoes.models import OOI, Reference
-from octopoes.models.ooi.network import IPPort, Protocol, PortState
-
+from octopoes.models.ooi.certificate import Certificate
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import (
+    IPPort,
+    Protocol,
+    PortState,
+    IPAddressV4,
+    IPAddressV6,
+    Network,
+)
+from octopoes.models.ooi.software import Software, SoftwareInstance
 from boefjes.job import NormalizerMeta
 
 
 def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI]:
     results = json.loads(raw)
     ooi = Reference.from_str(normalizer_meta.boefje_meta.input_ooi)
 
+    network = Network(name="internet").reference
+    ip = results["ip"]
+    ipvx = ipaddress.ip_address(ip)
+    if ipvx.version == 4:
+        ip_ooi = IPAddressV4(
+            address=ip,
+            network=network,
+        )
+    else:
+        ip_ooi = IPAddressV6(
+            address=ip,
+            network=network,
+        )
+
+    if 'dns' in results.keys() and 'names' in results["dns"].keys():
```

Comment:
Please use `in` and `not in` operators directly on an iterable or container type, for example:
`if "dns" in results: ...` rather than `if "dns" in results.keys(): ...`

Reply:
Thanks for the feedback ammar! The keys() approach must be a habit left by my java background ;-)